### PR TITLE
New Samples - IOC Restore, Workflow Manager

### PR DESCRIPTION
**Pull request by jshcodes**
# IOC Restore and Workflow Manager samples.
This update provides three new samples.

* IOC Restore
* Workflow Manager (terminal version)
* Workflow Manager (GUI version)

The ProxyTool sample is also retired as part of this update.

- [x] Documentation
- [x] Code sample

#### Unit test coverage
```shell
NOT REQUIRED FOR SAMPLE SUBMISSIONS
```

#### Bandit analysis
```shell
[main]	INFO	running on Python 3.13.0

Run started:2024-11-12 06:41:14.713795

Test results:
	No issues identified.

Code scanned:
	Total lines of code: 17996
	Total lines skipped (#nosec): 2

Run metrics:
	Total issues (by severity):
		Undefined: 0
		Low: 0
		Medium: 0
		High: 0
	Total issues (by confidence):
		Undefined: 0
		Low: 0
		Medium: 0
		High: 0
Files skipped (0):

```
## Added features and functionality
+ Added: IOC Restore sample.
    - `samples/ioc/ioc_restore.py`
    - Thanks go out to @am-cs-se for contributing this sample! 🙇 

+ Added: Workflow Manager (terminal version) and Workflow Manager (GUI version) samples.
    - `samples/workflows/workflow_manager.py`
    - `samples/workflows/workflow_manager_gui.py`

## Other
+ Retired: ProxyTool sample.
    - `samples/proxytool/*`

### CODE REVIEW

File: samples/ioc/ioc_restore.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+            raise SystemExit("Invalid Host Group name specified.")
+        self.host_group = hg_id[0]
+
+    def check_ioc_exists(self, ioc_type, ioc_value):
+        """Check if an IOC already exists."""
+        filter_query = f"type:'{ioc_type}'+value:'{ioc_value}'+deleted:false"
+        response: Result = self.api.indicator_search(filter=filter_query)
+        returned = False
+        if response.status_code == 200:
+            returned = bool(len(response.data) > 0)
```
Comment (crowdstrikedcs) [inline samples/ioc/ioc_restore.py:65]: bool(response.data)

File: samples/workflows/workflow_manager.py [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+                      default="auto"
+                      )
+
+    return parser.parse_args()
+
+
+def get_workflows(sdk: Workflows):
+    """Print a list of workflows within the tenant to the screen."""
+    workflow_list = []
+    pop_keys = ["trigger", "actions", "description", "conditions", "loops"]
```
Comment (crowdstrikedcs) [inline samples/workflows/workflow_manager.py:160]: move these constants up?

File: samples/workflows/workflow_manager.py [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+            workflow.pop(key)
+        workflow_list.append(workflow)
+
+    return workflow_list
+
+
+def get_executions(sdk: Workflows, definition_id: str):
+    """Retrieve all executions for the specified workflow."""
+    execution_list = []
+    pop_keys = ["definition_id", "definition_version", "ancestor_executions",
```
Comment (crowdstrikedcs) [inline samples/workflows/workflow_manager.py:181]: move these constants up?

**Review by crowdstrikedcs — APPROVED** [review]

File: samples/workflows/workflow_manager.py (hunk 2 above)
Comment (jshcodes) [reply samples/workflows/workflow_manager.py:160]: Unique to the method in this instance

File: samples/workflows/workflow_manager.py (hunk 3 above)
Comment (jshcodes) [reply samples/workflows/workflow_manager.py:181]: Unique to the method in this instance

**Pull request merged**